conda/env/env.py

```diff
@@ -207,7 +219,400 @@ def add(self, package_name):
+        :return: YAML string (if no stream specified), otherwise None
+        """
+        out = yaml_safe_dump(self.serialize(), stream)
+        if stream is None:
+            return out
+        return None
+
+    def to_file(self, filename: os.PathLike) -> None:
+        """Serialize the environment and dump it to a file as yaml.
+
+        :param filename: Name of the file to write the environment to
+        """
+        with open(filename, "w") as f:
+            yaml_safe_dump(self.serialize(), stream=f)
+
+    def __repr__(self) -> str:
+        groups_lines = []
+        for name, group in self.groups.items():
+            groups_lines.append(f"{name}:")
+            groups_lines.append(textwrap.indent(repr(group), "  "))
+
+        lines = [
+            f"name: {self.name if self.name else '<none>'}",
+            "options:",
+            textwrap.indent(
+                "\n".join([f"{key}: {value}" for key, value in self.options.items()]),
+                prefix="  ",
+            ),
+            "configuration:",
+            textwrap.indent(repr(self.config), "  "),
+            "requirements:",
+            textwrap.indent(repr(self.requirements), "  "),
+            "groups:",
+            textwrap.indent(
+                "\n".join(groups_lines),
+                "  ",
+            ),
+        ]
+        return "\n".join(lines)
+
+    def to_dict(self) -> dict:
+        """Return the environment as a dictionary.
+
+        :return: The environment as a dictionary
+        """
+        return self.serialize()
+
+    @classmethod
+    def from_history(cls, prefix: os.PathLike) -> EnvironmentV2:
+        """Create an Environment from the history of the requested packages in a prefix.
+
+        :param prefix: Prefix of the target conda environment
+        :return: EnvironmentV2 object containing all the user-requested conda packages
+        """
+        name = get_env_name(prefix)
+
+        requirements = []
+        for spec in History(prefix).get_requested_specs_map().values():
+            requirements.append(str(spec))
+
+        return EnvironmentV2.from_dict(
+            {
+                "name": name if name else None,
+                "requirements": requirements,
+                "variables": PrefixData(
+                    prefix, pip_interop_enabled=True
+                ).get_environment_env_vars(),
+            }
+        )
+
+    @classmethod
+    def from_prefix(cls, prefix: os.PathLike) -> EnvironmentV2:
+        """Create an Environment from the prefix of an environment on disk.
+
+        :param prefix: Prefix of the target conda environment
+        :return: EnvironmentV2 object containing all the conda packages in the prefix
+        """
+        pfd = PrefixData(prefix, pip_interop_enabled=True)
+        variables = pfd.get_environment_env_vars()
+
+        precs = tuple(PrefixGraph(pfd.iter_records()).graph)
+        grouped_precs = groupby(lambda x: x.package_type, precs)
```

Comment:
Why grouped? The toposort above is enough imo.

Reply:
No idea, this is just lifted from `from_environment`. I'll simplify this now.